File: code_examples/5-Holding_Data/4-Associative_Array/CSharp/main.cs
/* Created by: Mr. Coxall
 * Created on: Sep 2020
 * This program uses a dictionary
*/

using System;
using System.Collections.Generic;

/*
 * The Program class
*/
class Program {
    static void Main() {
        // this function uses a dictionary
        string airportName;

        // an empty dictionary
        Dictionary<string, string> airports = new Dictionary<string, string>();

        // a dictionary filled with data
        Dictionary<char, int> someInfo = new Dictionary<char, int>() {
            {'a', 1},
            {'b', 2},
            {'c', 3}
        };

        // adding items
        airports["YYZ"] = "Toronto Pearson";
        airports["YOW"] = "Ottawa Canada";
        airports["DUB"] = "Dublin Ireland";
        airports["LHR"] = "London Heathrow";

        Console.WriteLine("All the airports:");
        foreach (var pair in airports) {
            Console.WriteLine("The airport code is " + pair.Key + " for " + pair.Value);
        }
        Console.WriteLine();

        Console.Write("Type in an airport code: ");
        airportName = Console.ReadLine();
        Console.WriteLine();

        if (airports.ContainsKey(airportName)) {
            Console.WriteLine("The name of the airport you chose is " + airports[airportName]);
        } else {
            Console.WriteLine("That airport is not in the airport's dictionary.");
        }

    Console.WriteLine("\nDone.");
    }
}

File: code_examples/5-Holding_Data/4-Associative_Array/JavaScript/main.js
/* Created by: Mr. Coxall
 * Created on: Sep 2020
 * This program uses a map
 */

const prompt = require("prompt-sync")()

// this function uses a map
let airportName;

// an empty map
const airports = new Map();

// a map filled with data
const someInfo = new Map([
  ['a', 1],
  ['b', 2],
  ['c', 3],
]);

// adding items
airports.set("YYZ", "Toronto Pearson");
airports.set("YOW", "Ottawa Canada");
airports.set("DUB", "Dublin Ireland");
airports.set("LHR", "London Heathrow");

console.log("All the airports:");
for (let [code, name] of airports) {
  console.log(`The airport code is ${code} for ${name}`);
}
console.log();

airportName = prompt("Type in an airport code: ");
console.log();

if (airports.has(airportName)) {
  console.log("The name of the airport you chose is", airports.get(airportName));
} else {
  console.log("That airport is not in the airport's dictionary.");
}

console.log("\nDone.")
